src/tests/test_magic.py

```diff
@@ -252,6 +258,13 @@ def test_displaylimit(ip):
     assert "Shakespeare" not in result._repr_html_()
 
 
+@pytest.mark.parametrize("config_value, expected_length", [(3, 3), (6, 6)])
+def test_displaylimit_enabled(ip, config_value, expected_length):
+    ip.run_cell(f"%config SqlMagic.displaylimit = {expected_length}")
+    out = runsql(ip, "SELECT * FROM number_table;")
```

Comment:
I think we can test some more complex queries : e.g filters, store, schema, limit, etc

Reply:
Detect containing limit keyword:
I think this needs to be reconsidered since:
1. MS SQL doesn't natively support `LIMIT` keyword. Instead, they use something like `OFFSET X ROWS FETCH NEXT X ROWS ONLY`. Ref https://stackoverflow.com/questions/603724/how-to-implement-limit-with-sql-server
2. The query `%sql select * from "penguins.csv" limit 20` for me is almost equivalent to `%sql select * from "penguins.csv"`, since they mean The length of query result is large than `displaylimit`.

I think if we detect the query statement might be over-engineering this

Reply:
Is any specific complex queries you want to test with? Since I don't know what other use cases might affect by this change now

Reply:
e.g. make sure the following return the expected number of rows

With limit (> default, < limit, = limit)

`SELECT * FROM table_name LIMIT 3`

With conditions
`SELECT * FROM table_name WHERE ...`

Store (with limit <, >, == or without limit)
```
%%sql --save no_nulls --no-execute
SELECT *
FROM penguins.csv
WHERE body_mass_g IS NOT NULL and
sex IS NOT NULL

%%sql --with no_nulls
select * from no_nulls
```